hadoop-ozone/ozone-manager/src/main/java/org/apache/hadoop/ozone/om/KeyManagerImpl.java

```diff
@@ -1228,6 +1229,12 @@ private OmKeyInfo createFakeDirIfShould(String volume, String bucket,
         LOG.debug("Fake dir {} required for {}", targetKey, cacheKey);
         return createDirectoryKey(cacheValue.getCacheValue(), dirKey);
       }
+      // deletedKeys may contain deleted entry while iterating cache iterator
+      // To avoid race condition of flush of cache while iterating
+      // table iterator.
+      if (!exists) {
+        deletedKeys.add(cacheKey);
+      }
     }
```

Comment:
This is dumping a lot of keyTable cache entries into the temp set `deletedKeys`. And this method is getting called on every single `getFileStatus` / `getOzoneFileStatus`.

Is there a better way to fix this?

It might be fine for now since keyTable cache should only hold a few unflushed entries.

Reply:
Thanks @smengcl for your review. Another way to fix is this #5093 PR. Here a `table.isExist()` API being used, but my only concern is that cache is always faster and `isExist` is being called in #5093 PR till we find the target key as well it is being called inside `getNextKey() `call. Pls have look #5093 PR

Reply:
@smengcl Kindly advise.

Reply:
Alright. Let's merge this PR now for a fix. We could improve this further in #5093 .